gooddata-java/src/test/java/com/gooddata/sdk/service/hierarchicalconfig/HierarchicalConfigServiceAT.java

```diff
@@ -0,0 +1,126 @@
+import static org.hamcrest.Matchers.*;
+import static org.hamcrest.Matchers.is;
+import static org.testng.Assert.fail;
+
+public class HierarchicalConfigServiceAT extends AbstractGoodDataAT {
+
+    private static final String FIRST_SETTING_ITEM = "flag1";
+    private static final String SECOND_SETTING_ITEM = "flag2";
+
+    @Test(groups = "projectSettings", dependsOnGroups = "project")
+    public void createProjectSetting() {
+        final SettingItem settingItem = gd.getHierarchicalConfigService()
+                .upsertProjectSetting(project, new SettingItem(FIRST_SETTING_ITEM, "true", "project"));
+        final SettingItem secondSettingItem = gd.getHierarchicalConfigService()
+                .upsertProjectSetting(project, new SettingItem(SECOND_SETTING_ITEM, "false", "project"));
+
+        checkSettingItem(settingItem, FIRST_SETTING_ITEM, "true");
+        checkSettingItem(secondSettingItem, SECOND_SETTING_ITEM, "false");
+    }
+
+    @Test(groups = "projectSettings", dependsOnGroups = "project", expectedExceptions = GoodDataException.class)
+    public void createClientSettingViaProjectsApi() {
+        gd.getHierarchicalConfigService()
+                .upsertProjectSetting(project, new SettingItem(FIRST_SETTING_ITEM, "true", "client"));
+    }
+
+    @Test(groups = "projectSettings", dependsOnMethods = "createProjectSetting")
+    public void projectSettings() {
+        final Settings settings = gd.getHierarchicalConfigService().projectSettings(project);
+
+        assertThat(settings, hasItems(
+                new SettingItem(SECOND_SETTING_ITEM, "false", "project"),
+                new SettingItem(FIRST_SETTING_ITEM, "true", "project")));
+    }
+
+    @Test(groups = "projectSettings", dependsOnMethods = "projectSettings")
+    public void getProjectSetting() {
+        final SettingItem settingItem =
+                gd.getHierarchicalConfigService().getProjectSetting(project, SECOND_SETTING_ITEM);
+        checkSettingItem(settingItem, SECOND_SETTING_ITEM, "false");
+    }
+
+    @Test(groups = "projectSettings", dependsOnMethods = "getProjectSetting")
+    public void updateProjectSetting() {
+        final SettingItem settingItem = gd.getHierarchicalConfigService().getProjectSetting(project,
+                FIRST_SETTING_ITEM);
+
+        // update setting value
+        settingItem.setValue("false");
+        final SettingItem disabledSettingItem = gd.getHierarchicalConfigService().upsertProjectSetting(project, settingItem);
+        checkSettingItem(disabledSettingItem, FIRST_SETTING_ITEM, "false");
+
+        // enable again
+        settingItem.setValue("true");
+        final SettingItem enabledSettingItem = gd.getHierarchicalConfigService().upsertProjectSetting(project, settingItem);
+        checkSettingItem(enabledSettingItem, FIRST_SETTING_ITEM , "true");
+    }
+
+    @Test(groups = "projectSettings", dependsOnMethods = "updateProjectSetting")
+    public void changeProjectSetting() {
```

Comment:
Isn't this covered by previous method already?

Reply:
Truth, removed.